src/btree/row_modify.c

```diff
@@ -206,6 +206,8 @@ __wt_row_modify(WT_CURSOR_BTREE *cbt, const WT_ITEM *key, const WT_ITEM *value,
           session, page, cbt->ins_head, cbt->ins_stack, &ins, ins_size, skipdepth, exclusive));
     }
 
+    inserted_to_update_chain = true;
```

Comment:
This logic ensures to free the update only if it is added to update chain. How about we have allocated update but not successful in adding it to update chain, for eg, __wt_txn_modify at line 110 returns error, in that case we will not free the update, which could result in memory leak. Did I miss something ?

Reply:
No, it's the opposite. This is to ensure we only free the update if it is failed to be added to the update chain.

Reply:
So if we fail before this point. Update will be freed.